kahuna/public/js/edits/image-editor.js

```diff
@@ -115,6 +121,74 @@ imageEditor.controller('ImageEditorCtrl', [
             editsService.update(resource, data, image);
         });
     }
+
+    ctrl.collectionError = false;
+
+    ctrl.selectionMode = true;
+
+    ctrl.openCollectionTree = () => {
+        ctrl.addCollection = true;
+
+        if (!ctrl.collections) {
+            collections.getCollections().then(collections => {
+                ctrl.collections = collections.data.children;
```

Comment:
Sounds good - can you vary up how often the result gets reset? What happens if you're trying to add to collection someone else recently created?